### DISCUSSION TITLE
Installing latest python also changes the default python3 symlink

### DISCUSSION THREAD

**Discussion by dpetka2001** [jdx/mise#603] [Q&A]
First time `rtx` user, so please bear with me if I understood something wrong. On my system `python3 -V` gives version `3.10.6`. After trying to install latest python version with rtx by doing `rtx install python@latest`, it installs `3.11.3` but also seems to change to that version for the python3 symlink. If I try `python3 -V` afterwards it gives `3.11.3` instead of `3.10.6` like before rtx installation. Is this how it's supposed to work? I thought that it would just install latest version without changing the `python3` symlink as well. Thus when I would like to switch between the 2 versions, I would do `rtx use` or `rtx use -g` accordingly if I wanted a global or local to a directory configuration. Am I missing something maybe from my understanding or is this the intended behavior? Maybe I should configure something differently, so that it doesn't change the `python3` symlink on its own when installing latest Python?

**Comment by jdx** [comment]
rtx doesn't "change symlinks", please read the [beginner's guide](https://dev.to/jdxcode/beginners-guide-to-rtx-ac4) to understand what rtx does and what it is used for.

**Reply by dpetka2001** [reply]
So if I install `python@latest` it makes new binaries in `installs/python/latest/bin` and also new `python3` symlink in that directory which points to the python binary installed in that directory. The guide also mentions that `rtx` overrides any binaries that are on the system. So, is there no way for the 2 to co-exist? I would like for the `python3` system binary to take precedence over the rtx `python3` version and only use the rtx version when i do `rtx use python@3.11` in a directory. Is that something that doesn't align with rtx's philosophy? Until now i kept the system python intact (3.10) and only used other versions (3.11) in a virtualenv, so as to not break any python dependencies the system might need.

**Reply by jdx** [reply]
it sounds like maybe `rtx shell` is what you want instead of `rtx use`

**Reply by dpetka2001** [reply]
Thank you for your response. I will fiddle with it to see if it meets my needs and try to become more familiar with `rtx` and how I might have to adjust my workflow.
